### PR TITLE
feat: add config parameters for burn block start

### PR DESCRIPTION
**Pull request by obycode**
### Description

Added configuration parameters to customize the burn block at which to start processing Stacks blocks, when running on testnet or regtest.
  ```
  [burnchain]
  first_burn_block_height = 2582526
  first_burn_block_timestamp = 1710780828
  first_burn_block_hash = "000000000000001a17c68d43cb577d62074b63a09805e4a07e829ee717507f66"
  ```

### Applicable issues

- N/A

### Checklist

- [ ] Test coverage for new or modified code paths
- [x] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Review by kantai — APPROVED** [review]
LGTM!

**Review by kantai — APPROVED** [review]

**Review by hstove — DISMISSED** [review]

**Review by kantai — DISMISSED** [review]

**Review by wileyj — DISMISSED** [review]

**Review by hstove — APPROVED** [review]

File: testnet/stacks-node/src/config.rs [hunk 1]
```diff
@@ -513,8 +513,8 @@ impl Config {
                 .find(|epoch| epoch.epoch_id == StacksEpochId::Epoch10)
             {
                 assert!(
-                    epoch.start_height < burnchain.first_block_height,
-                    "FATAL: Epoch 1.0 start height must be before the first block height"
+                    epoch.start_height <= burnchain.first_block_height,
+                    "FATAL: Epoch 1.0 start height must be at or before the first block height"
```
[the remark this answers is not in the document]
Comment (obycode) [reply testnet/stacks-node/src/config.rs:517]: Yes, that's the case that it is allowing for. Should I be more explicit?

[the remark this answers is not in the document]
Comment (obycode) [reply testnet/stacks-node/src/config.rs:517]: Maybe a comment like this?
```rust
                // Epoch 1.0 start height can be equal to the first block height iff epoch 2.0
                // start height is also equal to the first block height.
```

[the remark this answers is not in the document]
Comment (CharlieC3) [reply testnet/stacks-node/src/config.rs:517]: IIRC If I try setting epoch 1.0 to anything other than `0`, the node exits and/or crashes.

**Review by kantai — DISMISSED** [review]

**Review by wileyj — DISMISSED** [review]

**Review by hstove — APPROVED** [review]

**Review by kantai — DISMISSED** [review]

**Review by wileyj — DISMISSED** [review]

**Review by kantai — APPROVED** [review]

**Comment by obycode** [comment]
These failing tests were fixed in #4556.

**Review by CharlieC3 — APPROVED** [review]

**Pull request merged**